File: NewsAlarm/app/src/main/java/com/alarm/newsalarm/outputmanager/SoundPlayer.java
package com.alarm.newsalarm.outputmanager;

import android.content.Context;
import android.media.AudioAttributes;
import android.media.MediaPlayer;

import com.alarm.newsalarm.R;
import com.alarm.newsalarm.utils.LogUtil;

public class SoundPlayer {

    private static final String CLASS_NAME = "SoundPlayer";
    private final Context context;
    private AudioAttributes attr;
    private MediaPlayer player;

    public SoundPlayer(Context context) {
        this.context = context;
        init();
    }

    private void init() {
        initAudioAttributes();
        initMediaPlayer();
    }

    private void initAudioAttributes() {
        attr = new AudioAttributes.Builder()
            .setUsage(AudioAttributes.USAGE_ALARM)
            .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
            .build();
    }

    private void initMediaPlayer() {
        player = MediaPlayer.create(context, R.raw.morningkiss, attr, 1);
        player.setVolume(0.15f, 0.15f);
        player.setLooping(true);
    }

    public void playBgm() {
        player.start();
        LogUtil.logD(CLASS_NAME, "playBgm", "sound player started to play!");
    }

    public void release() {
        if (player != null) {
            player.stop();
            player.release();
            player = null;
        }
        LogUtil.logD(CLASS_NAME, "release", "sound player released completely!");
    }
}

File: NewsAlarmKt/app/src/main/java/com/alarm/newsalarmkt/outputmanager/SoundPlayer.kt
package com.alarm.newsalarmkt.outputmanager

import android.content.Context
import android.media.AudioAttributes
import android.media.MediaPlayer
import com.alarm.newsalarmkt.R
import com.alarm.newsalarmkt.utils.LogUtil

class SoundPlayer(private val context: Context) {

    private lateinit var attr: AudioAttributes
    private var player: MediaPlayer? = null

    init {
        initAudioAttributes()
        initMediaPlayer()
    }

    private fun initAudioAttributes() {
        attr = AudioAttributes.Builder()
            .setUsage(AudioAttributes.USAGE_ALARM)
            .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
            .build()
    }

    private fun initMediaPlayer() {
        player = MediaPlayer.create(context, R.raw.morningkiss, attr, 1)
        player?.setVolume(0.15f, 0.15f)
        player?.isLooping = true
    }

    fun playBgm() {
        player?.start()
        LogUtil.logD(CLASS_NAME, "playBgm", "sound player started to play!")
    }

    fun release() {
        player?.stop()
        player?.release()
        player = null
        LogUtil.logD(CLASS_NAME, "release", "sound player released completely!")
    }

    companion object {
        private const val CLASS_NAME = "SoundPlayer"
    }
}
